### PR TITLE
ca-certificates: check cert is trusted for SSL use

### PR DESCRIPTION
**Pull request by rinsuki**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

currently, only "Always Trust"ed (for everything) certificates (in keychain) are added to ca-certificates.

but ca-certificates is for verify SSL CAs, so it would be good to include certs that are trusted only as SSL CAs.

Note, I'm not sure about `security verify-cert`'s `-p` options are works on old macOS (since currently I dont have a old OS installs), so you might be want to test it on older OS.

### CODE REVIEW

**Comment by Bo98** [comment]
> Note, I'm not sure about `security verify-cert`'s `-p` options are works on old macOS (since currently I dont have a old OS installs), so you might be want to test it on older OS.

It seems to be in the manpage in El Capitan, which is the oldest macOS we support.

File: Formula/ca-certificates.rb [hunk 1]
```diff
@@ -69,7 +70,7 @@ def macos_post_install
         -l -L
         -c #{tmpfile.path}
       ]
-      verify_args << "-R" << "offline" if MacOS.version >= :high_sierra
+      verify_args << "-R" << "offline" << "-p" << "ssl" if MacOS.version >= :high_sierra
```
Comment (SMillerDev) [inline Formula/ca-certificates.rb:73]: Since it's supported on el capitan, this should be in the normal array.

**Comment by carlocab** [comment]
I think we used to do something like this (back when the CA cert bundle was packaged with OpenSSL, etc), and some users complained about it. I may be misremembering, though.

**Review by MikeMcQuaid — COMMENTED** [review]
> I think we used to do something like this (back when the CA cert bundle was packaged with OpenSSL, etc), and some users complained about it. I may be misremembering, though.

Would want this verified before we merge this.

**Comment by Bo98** [comment]
> I think we used to do something like this (back when the CA cert bundle was packaged with OpenSSL, etc), and some users complained about it. I may be misremembering, though.

There was something that broke GnuTLS but it wasn't this, and I later fixed it properly by removing non-SSL certificates from output (but I neglected to go as far as checking trust). Not sure if all that's what you're referring to.

On the surface this PR makes sense at least.

**Comment by rinsuki** [comment]
> I think we used to do something like this (back when the CA cert bundle was packaged with OpenSSL, etc), and some users complained about it. I may be misremembering, though.

I think it is probably #1072 and #71191?

**Comment by rinsuki** [comment]
... can someone review this?

**Review by carlocab — COMMENTED** [review]
I'm fine with this, but this needs rebasing since we now have a new version of `ca-certificates`.

**Comment by rinsuki** [comment]
rebased!

**Review by carlocab — APPROVED** [review]
Thanks!

**Pull request closed without merging**